body = <<-EOF
<meta name="citation_title" content="b&#237;ogo/ncbi: interfaces to NCBI services for the Go language">
<meta name="citation_author" content="Daniel Kortschak, R"><meta name="citation_author" content="L Adelson, David">
<meta name="citation_publication_date" content="2017//">
<meta name="citation_journal_title" content="The Journal of Open Source Software">
<meta name="citation_pdf_url" content="http://www.theoj.org/joss-papers/joss.00234/10.21105.joss.00234.pdf">
<meta name="citation_doi" content="10.21105/joss.00234">
<meta name="citation_issn" content="2475-9066">
<div class="accepted-paper">
  <h1>b&#237;ogo/ncbi: interfaces to NCBI services for the Go language</h1>
  <div class="columns links">
    <div class="column four-fifths" style="padding-bottom: 10px;">
      <strong>Authors</strong>
      <ul class="author-list">
            <li><a href="http://orcid.org/0000-0001-8295-2301" target="_blank">R Daniel Kortschak</a></li>
            <li><a href="http://orcid.org/0000-0003-2404-5636" target="_blank">David L Adelson</a></li>
            </ul>
    </div>
    <div class="one-third column">
      <span class="repo">Repository:<br /><a href="https://github.com/biogo/ncbi">Repository link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Paper:<br /><a href="http://www.theoj.org/joss-papers/joss.00234/10.21105.joss.00234.pdf">PDF link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Review:<br /><a href="https://github.com/openjournals/joss-reviews/issues/234">View review issue &raquo;</a></span>
    </div>

    <div class="one-third column" style="padding-top: 20px;">
      <span class="repo">DOI:<br /><a href="https://doi.org/10.21105/joss.00234">https://doi.org/10.21105/joss.00234</a></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Status badge:<br /><img src="http://joss.theoj.org/papers/10.21105/joss.00234/status.svg"></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Citation:<br />
      <small>Kortschak et al., (2017). b&#237;ogo/ncbi: interfaces to NCBI services for the Go language. <em>Journal of Open Source Software</em>, 2(18), 234, doi:10.21105/joss.00234</small>
    </div>
  </div>
  <div class="paper-body">
  <h1 id="summary">Summary</h1>
  <p>The National Center for Biotechnology Information makes available BLAST sequence similarity search <span class="citation">(Biotechnology Information (US) 2017a)</span> and health science database search through the Entrez service <span class="citation">(Biotechnology Information (US) 2005)</span>. In addition to an interactive web interface, BLAST and Entrez provide an application programmer interface to allow programmatic use of these services via the BLAST URL API <span class="citation">(Biotechnology Information (US) 2017b)</span> and the Entrez EUtilities <span class="citation">(Biotechnology Information (US) 2010)</span>. The BioPerl suite <span class="citation">(Stajich J.E. 2002)</span> provides access to the BLAST API via Bio::Tools::Run::StandAloneBlastPlus <span class="citation">(Fields C. 2017a)</span> and to Entrez via Bio::Tools::EUtilities <span class="citation">(Fields C. 2017b)</span>. Similarly, Biopython <span class="citation">(Cock P.J.A. 2009)</span> provides access via the NCBIWWW function in the Bio.Blast module and functions in Bio.Entrez for EUtilities. Packages within b&#237;ogo/ncbi provide Go application programmer interfaces to the NCBI BLAST and EUtilities services. The design of b&#237;ogo/ncbi is light weight, allowing the user to make use of the Go language's control structures and data types, rather than imposing a library-specific access approach. In addition to allowing remote BLAST searches, BioPerl and Biopython provide mechanisms to parse XML output from local BLAST search via BioPerl's Bio::SearchIO and Biopythons Bio.Blast NCBIXML. Because of the modular design of b&#237;ogo/ncbi, XML encoded output from local BLAST searches or remote downloads can be parsed using the Go standard library's XML package.</p>
  <h1 id="references" class="unnumbered">References</h1>
  <div id="refs" class="references">
  <div id="ref-Entrez">
  <p>Biotechnology Information (US), National Center for. 2005. &#8220;Entrez Programming Utilities Help.&#8221; <a href="https://www.ncbi.nlm.nih.gov/books/NBK3836/" class="uri">https://www.ncbi.nlm.nih.gov/books/NBK3836/</a>.</p>
  </div>
  <div id="ref-EUtilities">
  <p>&#8212;&#8212;&#8212;. 2010. &#8220;Entrez Programming Utilities Help.&#8221; <a href="https://www.ncbi.nlm.nih.gov/books/NBK25501/" class="uri">https://www.ncbi.nlm.nih.gov/books/NBK25501/</a>.</p>
  </div>
  <div id="ref-BLAST">
  <p>&#8212;&#8212;&#8212;. 2017a. &#8220;Basic Local Alignment Search Tool.&#8221; Accessed March 3. <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="uri">https://blast.ncbi.nlm.nih.gov/Blast.cgi</a>.</p>
  </div>
  <div id="ref-BLASTURLAPI">
  <p>&#8212;&#8212;&#8212;. 2017b. &#8220;Common Url Api.&#8221; Accessed March 3. <a href="http://www.ncbi.nlm.nih.gov/BLAST/Doc/urlapi.html" class="uri">http://www.ncbi.nlm.nih.gov/BLAST/Doc/urlapi.html</a>.</p>
  </div>
  <div id="ref-Biopython">
  <p>Cock P.J.A., et al. 2009. &#8220;Biopython: Freely Available Python Tools for Computational Molecular Biology and Bioinformatics.&#8221; <em>Bioinformatics</em> 25 (11): 1422&#8211;3. doi:<a href="https://doi.org/10.1093/bioinformatics/btp163">10.1093/bioinformatics/btp163</a>.</p>
  </div>
  <div id="ref-BioPerlRUN">
  <p>Fields C. 2017a. &#8220;Bio-Eutilities.&#8221; Accessed March 3. <a href="https://github.com/bioperl/bioperl-run" class="uri">https://github.com/bioperl/bioperl-run</a>.</p>
  </div>
  <div id="ref-BioPerlEUtilities">
  <p>&#8212;&#8212;&#8212;. 2017b. &#8220;Bio-Eutilities.&#8221; Accessed March 3. <a href="https://github.com/bioperl/Bio-EUtilities" class="uri">https://github.com/bioperl/Bio-EUtilities</a>.</p>
  </div>
  <div id="ref-BioPerl">
  <p>Stajich J.E., et al. 2002. &#8220;The Bioperl Toolkit: Perl Modules for the Life Sciences.&#8221; <em>Genome Research</em> 12 (10): 1611&#8211;8. doi:<a href="https://doi.org/10.1101/gr.361602">10.1101/gr.361602</a>.</p>
  </div>
  </div>
  </div>
</div>
EOF
