body = <<-EOF
<meta name="citation_title" content="khmer release v2.1: software for biological sequence analysis">
<meta name="citation_author" content="Standage, Daniel"><meta name="citation_author" content="yari, Ali"><meta name="citation_author" content="J. Cohen, Lisa"><meta name="citation_author" content="R. Crusoe, Michael"><meta name="citation_author" content="Head, Tim"><meta name="citation_author" content="Irber, Luiz"><meta name="citation_author" content="EK Joslin, Shannon"><meta name="citation_author" content="B. Kingsley, N."><meta name="citation_author" content="D. Murray, Kevin"><meta name="citation_author" content="Neches, Russell"><meta name="citation_author" content="Scott, Camille"><meta name="citation_author" content="Shean, Ryan"><meta name="citation_author" content="Steinbiss, Sascha"><meta name="citation_author" content="Sydney, Cait"><meta name="citation_author" content="Titus Brown, C.">
<meta name="citation_publication_date" content="2017/07/03">
<meta name="citation_journal_title" content="The Journal of Open Source Software">
<meta name="citation_pdf_url" content="http://www.theoj.org/joss-papers/joss.00272/10.21105.joss.00272.pdf">
<meta name="citation_doi" content="10.21105/joss.00272">
<meta name="citation_issn" content="2475-9066">
<div class="accepted-paper">
  <h1>khmer release v2.1: software for biological sequence analysis</h1>
  <div class="columns links">
    <div class="column four-fifths" style="padding-bottom: 10px;">
      <strong>Authors</strong>
      <ul class="author-list">
            <li><a href="http://orcid.org/0000-0003-0342-8531" target="_blank">Daniel Standage</a></li>
            <li><a href="http://orcid.org/0000-0003-0925-4886" target="_blank">Ali Aliyari</a></li>
            <li><a href="http://orcid.org/0000-0002-3600-7218" target="_blank">Lisa J. Cohen</a></li>
            <li><a href="http://orcid.org/0000-0002-2961-9670" target="_blank">Michael R. Crusoe</a></li>
            <li><a href="http://orcid.org/0000-0003-0931-3698" target="_blank">Tim Head</a></li>
            <li><a href="http://orcid.org/0000-0003-4371-9659" target="_blank">Luiz Irber</a></li>
            <li><a href="http://orcid.org/0000-0001-5470-1193" target="_blank">Shannon EK Joslin</a></li>
            <li><a href="http://orcid.org/0000-0001-8574-8592" target="_blank">N. B. Kingsley</a></li>
            <li><a href="http://orcid.org/0000-0002-2466-1917" target="_blank">Kevin D. Murray</a></li>
            <li><a href="http://orcid.org/0000-0002-2055-8381" target="_blank">Russell Neches</a></li>
            <li><a href="http://orcid.org/0000-0001-8822-8779" target="_blank">Camille Scott</a></li>
            <li><a href="http://orcid.org/" target="_blank">Ryan Shean</a></li>
            <li><a href="http://orcid.org/0000-0002-2151-0574" target="_blank">Sascha Steinbiss</a></li>
            <li><a href="http://orcid.org/" target="_blank">Cait Sydney</a></li>
            <li><a href="http://orcid.org/0000-0001-6001-2677" target="_blank">C. Titus Brown</a></li>
            </ul>
    </div>
    <div class="one-third column">
      <span class="repo">Repository:<br /><a href="https://github.com/dib-lab/khmer">Repository link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Paper:<br /><a href="http://www.theoj.org/joss-papers/joss.00272/10.21105.joss.00272.pdf">PDF link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Review:<br /><a href="https://github.com/openjournals/joss-reviews/issues/272">View review issue &raquo;</a></span>
    </div>

    <div class="one-third column" style="padding-top: 20px;">
      <span class="repo">DOI:<br /><a href="https://doi.org/10.21105/joss.00272">https://doi.org/10.21105/joss.00272</a></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Status badge:<br /><img src="http://joss.theoj.org/papers/10.21105/joss.00272/status.svg"></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Citation:<br />
      <small>Standage et al., (2017). khmer release v2.1: software for biological sequence analysis. <em>Journal of Open Source Software</em>, 2(15), 272, doi:10.21105/joss.00272</small>
    </div>
  </div>
  <div class="paper-body">
  <h1 id="summary">Summary</h1>
  <p>The khmer software is a set of command-line tools built around a Python library designed for analysis of large DNA sequence collections. Functionality in khmer has primarily been motivated by scaling issues with (meta)genome and (meta)transcriptome assembly <span class="citation">(Crusoe et al. 2015)</span>. khmer provides convenient access to several <em>k</em>-mer based operations on DNA sequence collections, such as abundance filtering, error trimming, assembly graph partitioning, and most notably, abundance normalization of reads <span class="citation">(C. T. Brown et al. 2012)</span> and streaming error trimming of reads <span class="citation">(Zhang, Awad, and Brown 2015)</span>. All of these operations utilize khmer's implementation of two primary data structures, the Bloom filter and the Count-Min Sketch, for efficient probabalistic storage of <em>k</em>-mer presence or <em>k</em>-mer abundance, respectively <span class="citation">(J. Pell et al. 2012; J. A. C.-K. Zhang Qingpeng AND Pell 2014)</span>.</p>
  <p>Release version 2.1 of the khmer software includes several new features that extend its utility to a wider set of sequence processing and analysis problems. These include the following: support for variable-coverage trimming of sequence reads; support for <em>k</em> &gt; 32 using the non-reversible hash function MurmurHash3 (https://github.com/aappleby/smhasher); a new optional Count-Min Sketch implementation providing increased storage efficiency; support for assembly directly from a <em>k</em>-mer graph; a script for computing a compact de Bruijn graph from a <em>k</em>-mer graph; and several examples of khmer's Python and C++ APIs for those interested in using and extending the library.</p>
  <h1 id="references" class="unnumbered">References</h1>
  <div id="refs" class="references">
  <div id="ref-brown2012">
  <p>Brown, C. Titus, Adina Howe, Qingpeng Zhang, Alexis B. Pyrkosz, and Timothy H. Brom. 2012. &#8220;A Reference-Free Algorithm for Computational Normalization of Shotgun Sequencing Data.&#8221;</p>
  </div>
  <div id="ref-crusoe2015">
  <p>Crusoe, MR, HF Alameldin, S Awad, E Boucher, A Caldwell, R Cartwright, A Charbonneau, et al. 2015. &#8220;The Khmer Software Package: Enabling Efficient Nucleotide Sequence Analysis.&#8221; <em>F1000Research</em> 4 (900). doi:<a href="https://doi.org/10.12688/f1000research.6924.1">10.12688/f1000research.6924.1</a>.</p>
  </div>
  <div id="ref-pell2012">
  <p>Pell, Jason, Arend Hintze, Rosangela Canino-Koning, Adina Howe, James M. Tiedje, and C. Titus Brown. 2012. &#8220;Scaling Metagenome Sequence Assembly with Probabilistic de Bruijn Graphs.&#8221; <em>Proceedings of the National Academy of Sciences</em> 109 (33): 13272&#8211;7. doi:<a href="https://doi.org/10.1073/pnas.1121464109">10.1073/pnas.1121464109</a>.</p>
  </div>
  <div id="ref-zhang2014">
  <p>Zhang, Jason AND Canino-Koning, Qingpeng AND Pell. 2014. &#8220;These Are Not the K-Mers You Are Looking for: Efficient Online K-Mer Counting Using a Probabilistic Data Structure.&#8221; <em>PLOS ONE</em> 9 (7). Public Library of Science: 1&#8211;13. doi:<a href="https://doi.org/10.1371/journal.pone.0101271">10.1371/journal.pone.0101271</a>.</p>
  </div>
  <div id="ref-zhang2015">
  <p>Zhang, Qingpeng, Sherine Awad, and C. Titus Brown. 2015. &#8220;Crossing the Streams: A Framework for Streaming Analysis of Short Dna Sequencing Reads.&#8221; <em>PeerJ PrePrints</em> 3 (March): e890v1. doi:<a href="https://doi.org/10.7287/peerj.preprints.890v1">10.7287/peerj.preprints.890v1</a>.</p>
  </div>
  </div>
  </div>
</div>
EOF
