body = <<-EOF
<meta name="citation_title" content="qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots">
<meta name="citation_author" content="D. Turner, Stephen">
<meta name="citation_publication_date" content="2018//">
<meta name="citation_journal_title" content="The Journal of Open Source Software">
<meta name="citation_pdf_url" content="http://www.theoj.org/joss-papers/joss.00731/10.21105.joss.00731.pdf">
<meta name="citation_doi" content="10.21105/joss.00731">
<meta name="citation_issn" content="2475-9066">
<div class="accepted-paper">
  <h1>qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots</h1>
  <div class="columns links">
    <div class="column four-fifths" style="padding-bottom: 10px;">
      <strong>Authors</strong>
      <ul class="author-list">
            <li><a href="http://orcid.org/0000-0001-9140-9028" target="_blank">Stephen D. Turner</a></li>
            </ul>
    </div>
    <div class="one-third column">
      <span class="repo">Repository:<br /><a href="https://github.com/stephenturner/qqman/">Repository link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Paper:<br /><a href="http://www.theoj.org/joss-papers/joss.00731/10.21105.joss.00731.pdf">PDF link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Review:<br /><a href="https://github.com/openjournals/joss-reviews/issues/731">View review issue &raquo;</a></span>
    </div>

    <div class="one-third column" style="padding-top: 20px;">
      <span class="repo">DOI:<br /><a href="https://doi.org/10.21105/joss.00731">https://doi.org/10.21105/joss.00731</a></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Status badge:<br /><img src="http://joss.theoj.org/papers/10.21105/joss.00731/status.svg"></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">
        Submitted: 25 April 2018 <br />
        Published: 19 May 2018
      </span>

    </div>
    <div class="two-thirds column" style="padding-top: 20px;">
      <span class="paper">Citation:<br />
      <small>Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. <em>Journal of Open Source Software</em>, 3(25), 731. https://doi.org/10.21105/joss.00731</small>
    </div>
  </div>
  <div class="paper-body">
  <h1 id="summary">Summary</h1>
  <p>Genome-wide association studies (GWAS) have been successful in identifying thousands of trait and disease-associated single nucleotide polymorphisms (SNPs). The primary result of a GWAS analysis is a list of SNPs, their associated chromosomal position, and a P-value representing the statistical significance of the association. A commonly used method used to visualize GWAS results is the &#8220;manhattan plot&#8221; &#8211; a plot of the <span class="math inline">&#8722;<em>l</em><em>o</em><em>g</em><sub>10</sub>(<em>P</em>)</span> of the association statistic on the <em>y</em>-axis versus the chromosomal position of the SNP on the <em>x</em>-axis. Another commonly used results diagnostic plot is the quantile-quantile (&#8220;Q-Q&#8221;) plot. Q-Q plots display the observed association <em>P</em>-value for all SNPs on the y-axis versus the expected uniform distribution of P-values under the null hypothesis of no association on the <em>x</em>-axis.</p>
  <p>One of the most commonly used software packages for manipulating and analyzing GWAS data is PLINK (<span class="citation" data-cites="purcell2007plink">Purcell et al. (2007)</span>). <code>qqman</code> is an R package that allows for quick and flexible generation of publication-ready Q-Q and manhattan plots directly from PLINK results files. The <code>qqman</code> package is a user-friendly tool to visualize results from GWAS experiments using Q-Q and manhattan plots. The <code>manhattan()</code> function in the <code>qqman</code> package takes a data frame with columns containing the chromosome number, chromosomal position, P-value, and optionally the SNP name. By default, <code>manhattan()</code> looks for column names corresponding to those output by the <code>plink --assoc</code> command, namely, &#8220;CHR,&#8221; &#8220;BP,&#8221; &#8220;P,&#8221; and &#8220;SNP,&#8221; although different column names can be specified by the user. Thresholds for suggestive and genome-wide significance are drawn, and users also have the ability to highlight/annotate SNPs of interest. Finally, the <code>qq()</code> function can be used to generate a Q-Q plot to visualize the distribution of association <em>P</em>-values. An example of the plots produced by <code>qqman</code> is shown in Figure 1.</p>
  <p>These graphics can be created in other software, such as the standalone desktop software Haploview (<span class="citation" data-cites="barrett2004haploview">Barrett et al. (2004)</span>), or for focused regions using the web-based application LocusZoom (<span class="citation" data-cites="pruim2010locuszoom">Pruim et al. (2010)</span>). Conversely, <code>qqman</code> is distributed as an R package with no other dependencies that can be easily integrated into existing R-based scripted workflows to further enable automated reproducible research. Furthermore, users can take advantage of R&#8217;s very granular control of graphical output, enabling a high degree of customizability in creating high-resolution, publication-ready figures. The <code>qqman</code> package ships with example data and a detailed vignette illustrating its usage and further features not described here. The package is available on GitHub under the GNU General Public License at <a href="https://github.com/stephenturner/qqman" class="uri">https://github.com/stephenturner/qqman</a> and on the Comprehensive R Archive Network (CRAN) at <a href="https://cran.r-project.org/package=qqman" class="uri">https://cran.r-project.org/package=qqman</a>.</p>
  <figure>
  <img src="fig1.png" alt="Manhattan plot highlighting SNPs of interest on chromosome 3, with Q-Q plot showing substantial deviation from the diagonal (inset)." /><figcaption>Manhattan plot highlighting SNPs of interest on chromosome 3, with Q-Q plot showing substantial deviation from the diagonal (inset).</figcaption>
  </figure>
  <h1 id="references" class="unnumbered">References</h1>
  <div id="refs" class="references">
  <div id="ref-barrett2004haploview">
  <p>Barrett, Jeffrey C, B Fry, JDMJ Maller, and Mark J Daly. 2004. &#8220;Haploview: Analysis and Visualization of Ld and Haplotype Maps.&#8221; <em>Bioinformatics</em> 21 (2). Oxford University Press:263&#8211;65. <a href="https://doi.org/10.1093/bioinformatics/bth457" class="uri">https://doi.org/10.1093/bioinformatics/bth457</a>.</p>
  </div>
  <div id="ref-pruim2010locuszoom">
  <p>Pruim, Randall J, Ryan P Welch, Serena Sanna, Tanya M Teslovich, Peter S Chines, Terry P Gliedt, Michael Boehnke, Gon&#231;alo R Abecasis, and Cristen J Willer. 2010. &#8220;LocusZoom: Regional Visualization of Genome-Wide Association Scan Results.&#8221; <em>Bioinformatics</em> 26 (18). Oxford University Press:2336&#8211;7. <a href="https://doi.org/10.1093/bioinformatics/btq419" class="uri">https://doi.org/10.1093/bioinformatics/btq419</a>.</p>
  </div>
  <div id="ref-purcell2007plink">
  <p>Purcell, Shaun, Benjamin Neale, Kathe Todd-Brown, Lori Thomas, Manuel AR Ferreira, David Bender, Julian Maller, et al. 2007. &#8220;PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses.&#8221; <em>The American Journal of Human Genetics</em> 81 (3). Elsevier:559&#8211;75. <a href="https://doi.org/10.1086/519795" class="uri">https://doi.org/10.1086/519795</a>.</p>
  </div>
  </div>
  </div>
</div>
EOF
