body = <<-EOF
<meta name="citation_title" content="reper: Genome-wide identification, classification and quantification of repetitive elements without an assembled genome">
<meta name="citation_author" content="Terhoeven, Niklas"><meta name="citation_author" content="Schultz, J&#246;rg"><meta name="citation_author" content="Hackl, Thomas">
<meta name="citation_publication_date" content="2018//">
<meta name="citation_journal_title" content="The Journal of Open Source Software">
<meta name="citation_pdf_url" content="http://www.theoj.org/joss-papers/joss.00527/10.21105.joss.00527.pdf">
<meta name="citation_doi" content="10.21105/joss.00527">
<meta name="citation_issn" content="2475-9066">
<div class="accepted-paper">
  <h1>reper: Genome-wide identification, classification and quantification of repetitive elements without an assembled genome</h1>
  <div class="columns links">
    <div class="column four-fifths" style="padding-bottom: 10px;">
      <strong>Authors</strong>
      <ul class="author-list">
            <li><a href="http://orcid.org/0000-0002-8669-0000" target="_blank">Niklas Terhoeven</a></li>
            <li><a href="http://orcid.org/" target="_blank">J&#246;rg Schultz</a></li>
            <li><a href="http://orcid.org/0000-0002-0022-320X" target="_blank">Thomas Hackl</a></li>
            </ul>
    </div>
    <div class="one-third column">
      <span class="repo">Repository:<br /><a href="https://github.com/nterhoeven/reper">Repository link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Paper:<br /><a href="http://www.theoj.org/joss-papers/joss.00527/10.21105.joss.00527.pdf">PDF link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Review:<br /><a href="https://github.com/openjournals/joss-reviews/issues/527">View review issue &raquo;</a></span>
    </div>

    <div class="one-third column" style="padding-top: 20px;">
      <span class="repo">DOI:<br /><a href="https://doi.org/10.21105/joss.00527">https://doi.org/10.21105/joss.00527</a></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Status badge:<br /><img src="http://joss.theoj.org/papers/10.21105/joss.00527/status.svg"></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">
        Submitted: 21 December 2017 <br />
        Published: 08 February 2018
      </span>

    </div>
    <div class="two-thirds column" style="padding-top: 20px;">
      <span class="paper">Citation:<br />
      <small>Terhoeven et al., (2018). reper: Genome-wide identification, classification and quantification of repetitive elements without an assembled genome. <em>Journal of Open Source Software</em>, 3(22), 527. https://doi.org/10.21105/joss.00527</small>
    </div>
  </div>
  <div class="paper-body">
  <h1 id="summary">Summary</h1>
  <p>Repetitive elements constitute a substantial fraction of most eukaryotic genomes. Still, their actual amount differs strongly between species. For example, the genome of <em>Saccharomyces cervisiae</em> contains only about 3 % repeats <span class="citation" data-cites="kim_transposable_1998">(Kim et al. 1998)</span>, <em>Arabidopsis</em> harbours 14 % <span class="citation" data-cites="the_arabidopsis_genome_initiative_analysis_2000">(The Arabidopsis Genome Initiative 2000)</span>, human 50 % <span class="citation" data-cites="lander_initial_2001">(Lander et al. 2001)</span> and wheat even 90 % <span class="citation" data-cites="clavijo_improved_2017">(Clavijo et al. 2017)</span>.</p>
  <p>Annotation and Classification of these elements is a pivotal step in the annotation of each genome. Furthermore, tracing their history can give ample insights into the evolution of a genome and thereby, of a species. Accordingly, different methods for repeat annotation have been developed <span class="citation" data-cites="smit_repeatmasker_2013 benson_tandem_1999 gymrek_lobstr:_2012">(A. Smit, Hubley, and Green 2013; Benson 1999; Gymrek et al. 2012)</span>. Still, typically they rely on an assembled genome sequence &#8211; a prerequisite which can lead to erroneous results. As repetitive elements are highly similar assembly algorithms will collapse repeat variants into a single occurrence or not assemble the repetitive regions at all. Thus, the annotation of repeat regions and thereby the characterization of their content and diversity solely based on an assembled genome sequence can give misleading results.</p>
  <p>To address this challenge, we developed reper, a kmer based method to detect, classify and quantify repeats in next generation sequencing (NGS) data without the need of a genome assembly. Our pipeline samples reads with high kmer coverage directly from the NGS dataset. The kmer counts are acquired using jellyfish <span class="citation" data-cites="marcais_fast_2011">(Mar&#231;ais and Kingsford 2011)</span>. This subset is assembled using the transcriptome assembler Trinity <span class="citation" data-cites="grabherr_full-length_2011">(Grabherr et al. 2011)</span>, allowing reper to recover repeat variants at a high resolution. To create exemplar sequences of each repeat in the genome, the assembled repeats are clustered using cd-hit <span class="citation" data-cites="li_cd-hit:_2006 fu_cd-hit:_2012">(W. Li and Godzik 2006; Fu et al. 2012)</span>. These are further classified based on homology to known repeats using multiple blast <span class="citation" data-cites="camacho_blast">(Camacho et al. 2009)</span> searches. Since reper was developed with a focus on plant data, the default classification libraries are REdat <span class="citation" data-cites="nussbaumer_mips_2012">(Nussbaumer et al. 2012)</span> for repeats, and refseq <span class="citation" data-cites="oleary_reference_2016">(O&#8217;Leary et al. 2016)</span> for chloroplast and mitochondrial sequences. The reference database, however, can easily be customized to the user&#8217;s needs. A configuration script for the popular, but proprietary database repbase is provided with the package as well. Next, the repeat content is quantified on sequence, cluster and class level based on read mappings using bowtie2 and samtools <span class="citation" data-cites="langmead_fast_2012 li_sequence_2009">(Langmead and Salzberg 2012; H. Li et al. 2009)</span>. Finally, the repeat landscape can be analyzed and graphically represented with the R script provided with the pipeline. Currently, reper is specifically customized to work with paired-end Illumina data, but support of long-read technologies such as PacBio and Nanopore is in development.</p>
  <p>To date, there is only a single software package with a similar functionality to reper, namely dnaPipeTE <span class="citation" data-cites="goubert_novo_2015">(Goubert et al. 2015)</span>. Still, it relies on dependencies like RepeatMasker, which has to be installed independently as well as the proprietary repeat database repbase by giri. Contrasting, The reper source code is available on <a href="https://github.com/nterhoeven/reper">github</a> under the MIT license. To further ease installation and usage, a Docker container with a complete reper installation is also provided. Since reper is usually run in an HPC environment where users don&#8217;t have root or Docker rights, we furthermore made a singularity image available which can be used with standard user permissions.</p>
  <p>We are currently using reper to analyze the repeat content in different plant genome sequencing projects. An example using <em>Beta vulgaris</em> data is given in the tutorial section of the <a href="https://github.com/nterhoeven/reper/wiki">reper wiki</a>.</p>
  <figure>
  <img src="workflow.png" alt="schematic overview of workflow" /><figcaption>schematic overview of workflow</figcaption>
  </figure>
  <h1 id="acknowledgements">Acknowledgements</h1>
  <p>We would like to thank Markus Ankenbrand and Frank F&#246;rster for valuable discussions and their support and advice on different topics like Docker and pipeline design.</p>
  <h1 id="references" class="unnumbered">References</h1>
  <div id="refs" class="references">
  <div id="ref-benson_tandem_1999">
  <p>Benson, G. 1999. &#8220;Tandem Repeats Finder: A Program to Analyze DNA Sequences.&#8221; <em>Nucleic Acids Research</em> 27 (2):573&#8211;80. <a href="https://doi.org/10.1093/nar/27.2.573" class="uri">https://doi.org/10.1093/nar/27.2.573</a>.</p>
  </div>
  <div id="ref-camacho_blast">
  <p>Camacho, Christiam, George Coulouris, Vahram Avagyan, Ning Ma, Jason Papadopoulos, Kevin Bealer, and Thomas L Madden. 2009. &#8220;BLAST+: Architecture and Applications.&#8221; <em>BMC Bioinformatics</em> 10 (1):421. <a href="https://doi.org/10.1186/1471-2105-10-421" class="uri">https://doi.org/10.1186/1471-2105-10-421</a>.</p>
  </div>
  <div id="ref-clavijo_improved_2017">
  <p>Clavijo, Bernardo J., Luca Venturini, Christian Schudoma, Gonzalo Garcia Accinelli, Gemy Kaithakottil, Jonathan Wright, Philippa Borrill, et al. 2017. &#8220;An Improved Assembly and Annotation of the Allohexaploid Wheat Genome Identifies Complete Families of Agronomic Genes and Provides Genomic Evidence for Chromosomal Translocations.&#8221; <em>Genome Research</em> 27 (5):885&#8211;96. <a href="https://doi.org/10.1101/gr.217117.116" class="uri">https://doi.org/10.1101/gr.217117.116</a>.</p>
  </div>
  <div id="ref-fu_cd-hit:_2012">
  <p>Fu, Limin, Beifang Niu, Zhengwei Zhu, Sitao Wu, and Weizhong Li. 2012. &#8220;CD-HIT: Accelerated for Clustering the Next-Generation Sequencing Data.&#8221; <em>Bioinformatics</em> 28 (23):3150&#8211;2. <a href="https://doi.org/10.1093/bioinformatics/bts565" class="uri">https://doi.org/10.1093/bioinformatics/bts565</a>.</p>
  </div>
  <div id="ref-goubert_novo_2015">
  <p>Goubert, Cl&#233;ment, Laurent Modolo, Cristina Vieira, Claire ValienteMoro, Patrick Mavingui, and Matthieu Boulesteix. 2015. &#8220;De Novo Assembly and Annotation of the Asian Tiger Mosquito (Aedes Albopictus) Repeatome with dnaPipeTE from Raw Genomic Reads and Comparative Analysis with the Yellow Fever Mosquito (Aedes Aegypti).&#8221; <em>Genome Biology and Evolution</em> 7 (4):1192&#8211;1205. <a href="https://doi.org/10.1093/gbe/evv050" class="uri">https://doi.org/10.1093/gbe/evv050</a>.</p>
  </div>
  <div id="ref-grabherr_full-length_2011">
  <p>Grabherr, Manfred G, Brian J Haas, Moran Yassour, Joshua Z Levin, Dawn A Thompson, Ido Amit, Xian Adiconis, et al. 2011. &#8220;Full-Length Transcriptome Assembly from RNA-Seq Data Without a Reference Genome.&#8221; <em>Nature Biotechnology</em> 29 (7):644&#8211;52. <a href="https://doi.org/10.1038/nbt.1883" class="uri">https://doi.org/10.1038/nbt.1883</a>.</p>
  </div>
  <div id="ref-gymrek_lobstr:_2012">
  <p>Gymrek, Melissa, David Golan, Saharon Rosset, and Yaniv Erlich. 2012. &#8220;lobSTR: A Short Tandem Repeat Profiler for Personal Genomes.&#8221; <em>Genome Research</em> 22 (6):1154&#8211;62. <a href="https://doi.org/10.1101/gr.135780.111" class="uri">https://doi.org/10.1101/gr.135780.111</a>.</p>
  </div>
  <div id="ref-kim_transposable_1998">
  <p>Kim, Jin M., Swathi Vanguri, Jef D. Boeke, Abram Gabriel, and Daniel F. Voytas. 1998. &#8220;Transposable Elements and Genome Organization: A Comprehensive Survey of Retrotransposons Revealed by the Complete <em>Saccharomyces Cerevisiae</em> Genome Sequence.&#8221; <em>Genome Research</em> 8 (5):464&#8211;78. <a href="https://doi.org/10.1101/gr.8.5.464" class="uri">https://doi.org/10.1101/gr.8.5.464</a>.</p>
  </div>
  <div id="ref-lander_initial_2001">
  <p>Lander, Eric S., Lauren M. Linton, Bruce Birren, Chad Nusbaum, Michael C. Zody, Jennifer Baldwin, Keri Devon, et al. 2001. &#8220;Initial Sequencing and Analysis of the Human Genome.&#8221; <em>Nature</em> 409 (6822):860&#8211;921. <a href="https://doi.org/10.1038/35057062" class="uri">https://doi.org/10.1038/35057062</a>.</p>
  </div>
  <div id="ref-langmead_fast_2012">
  <p>Langmead, Ben, and Steven L Salzberg. 2012. &#8220;Fast Gapped-Read Alignment with Bowtie 2.&#8221; <em>Nature Methods</em> 9 (4):357&#8211;59. <a href="https://doi.org/10.1038/nmeth.1923" class="uri">https://doi.org/10.1038/nmeth.1923</a>.</p>
  </div>
  <div id="ref-li_sequence_2009">
  <p>Li, H., B. Handsaker, A. Wysoker, T. Fennell, J. Ruan, N. Homer, G. Marth, G. Abecasis, R. Durbin, and 1000 Genome Project Data Processing Subgroup. 2009. &#8220;The Sequence Alignment/Map Format and SAMtools.&#8221; <em>Bioinformatics</em> 25 (16):2078&#8211;9. <a href="https://doi.org/10.1093/bioinformatics/btp352" class="uri">https://doi.org/10.1093/bioinformatics/btp352</a>.</p>
  </div>
  <div id="ref-li_cd-hit:_2006">
  <p>Li, W., and A. Godzik. 2006. &#8220;Cd-Hit: A Fast Program for Clustering and Comparing Large Sets of Protein or Nucleotide Sequences.&#8221; <em>Bioinformatics</em> 22 (13):1658&#8211;9. <a href="https://doi.org/10.1093/bioinformatics/btl158" class="uri">https://doi.org/10.1093/bioinformatics/btl158</a>.</p>
  </div>
  <div id="ref-marcais_fast_2011">
  <p>Mar&#231;ais, Guillaume, and Carl Kingsford. 2011. &#8220;A Fast, Lock-Free Approach for Efficient Parallel Counting of Occurrences of K-Mers.&#8221; <em>Bioinformatics</em> 27 (6):764&#8211;70. <a href="https://doi.org/10.1093/bioinformatics/btr011" class="uri">https://doi.org/10.1093/bioinformatics/btr011</a>.</p>
  </div>
  <div id="ref-nussbaumer_mips_2012">
  <p>Nussbaumer, Thomas, Mihaela M. Martis, Stephan K. Roessner, Matthias Pfeifer, Kai C. Bader, Sapna Sharma, Heidrun Gundlach, and Manuel Spannagl. 2012. &#8220;MIPS PlantsDB: A Database Framework for Comparative Plant Genome Research.&#8221; <em>Nucleic Acids Research</em> 41 (D1):D1144&#8211;D1151. <a href="https://doi.org/10.1093/nar/gks1153" class="uri">https://doi.org/10.1093/nar/gks1153</a>.</p>
  </div>
  <div id="ref-oleary_reference_2016">
  <p>O&#8217;Leary, Nuala A., Mathew W. Wright, J. Rodney Brister, Stacy Ciufo, Diana Haddad, Rich McVeigh, Bhanu Rajput, et al. 2016. &#8220;Reference Sequence (RefSeq) Database at NCBI: Current Status, Taxonomic Expansion, and Functional Annotation.&#8221; <em>Nucleic Acids Research</em> 44 (D1):D733&#8211;D745. <a href="https://doi.org/10.1093/nar/gkv1189" class="uri">https://doi.org/10.1093/nar/gkv1189</a>.</p>
  </div>
  <div id="ref-smit_repeatmasker_2013">
  <p>Smit, AFA, R Hubley, and P Green. 2013. &#8220;RepeatMasker Open-4.0.&#8221; <a href="http://www.repeatmasker.org" class="uri">http://www.repeatmasker.org</a>.</p>
  </div>
  <div id="ref-the_arabidopsis_genome_initiative_analysis_2000">
  <p>The Arabidopsis Genome Initiative. 2000. &#8220;Analysis of the Genome Sequence of the Flowering Plant Arabidopsis Thaliana.&#8221; <em>Nature</em> 408 (6814):796&#8211;815. <a href="https://doi.org/10.1038/35048692" class="uri">https://doi.org/10.1038/35048692</a>.</p>
  </div>
  </div>
  </div>
</div>
EOF
