body = <<-EOF
<meta name="citation_title" content="genomepy: download genomes the easy way">
<meta name="citation_author" content="J. van Heeringen, Simon">
<meta name="citation_publication_date" content="2017/08/15">
<meta name="citation_journal_title" content="The Journal of Open Source Software">
<meta name="citation_pdf_url" content="http://www.theoj.org/joss-papers/joss.00320/10.21105.joss.00320.pdf">
<meta name="citation_doi" content="10.21105/joss.00320">
<meta name="citation_issn" content="2475-9066">
<div class="accepted-paper">
  <h1>genomepy: download genomes the easy way</h1>
  <div class="columns links">
    <div class="column four-fifths" style="padding-bottom: 10px;">
      <strong>Authors</strong>
      <ul class="author-list">
            <li><a href="http://orcid.org/0000-0002-0411-3219" target="_blank">Simon J. van Heeringen</a></li>
            </ul>
    </div>
    <div class="one-third column">
      <span class="repo">Repository:<br /><a href="http://github.com/simonvh/genomepy">Repository link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Paper:<br /><a href="http://www.theoj.org/joss-papers/joss.00320/10.21105.joss.00320.pdf">PDF link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Review:<br /><a href="https://github.com/openjournals/joss-reviews/issues/320">View review issue &raquo;</a></span>
    </div>

    <div class="one-third column" style="padding-top: 20px;">
      <span class="repo">DOI:<br /><a href="https://doi.org/10.21105/joss.00320">https://doi.org/10.21105/joss.00320</a></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Status badge:<br /><img src="http://joss.theoj.org/papers/10.21105/joss.00320/status.svg"></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Citation:<br />
      <small>Heeringen, (2017). genomepy: download genomes the easy way. <em>Journal of Open Source Software</em>, 2(16), 320, doi:10.21105/joss.00320</small>
    </div>
  </div>
  <div class="paper-body">
  <h1 id="summary">Summary</h1>
  <p>The technical advances in high-throughput DNA sequencing have resulted in data sets of increasing size and complexity. It is essential that experiments are analyzed in a reproducible manner. Computational workflow systems allow for automation of analysis pipelines, that scale from personal computers to HPC and cloud environments <span class="citation">(Koster and Rahmann 2012; Di Tommaso et al. 2017; Vivian et al. 2017)</span>. While standardized solutions exist for installation of software and analysis tools <span class="citation">(&#8220;Pip,&#8221; n.d.; &#8220;Bioconda,&#8221; n.d.)</span>, data download often has to either be performed manually or be scripted on a case-by-case basis.</p>
  <p>In many analysis workflows, sequencing reads will need to be mapped to a reference genome. Here we present genomepy, a simple software package to automate the download of genomic sequences. It contains both command-line tools as well as a Python API. Supported providers for genomes include UCSC, NCBI and Ensembl. Downloaded genome sequences can be soft- or hard-masked and specific chromosomes or scaffolds can be either included or excluded based on regular expressions. Genomepy is free and open source software and can be installed through standard package managers <span class="citation">(&#8220;Pip,&#8221; n.d.; &#8220;Bioconda,&#8221; n.d.)</span>.</p>
  <p>In short, genomepy enables simple, straightforward and automatic downloads of genomic sequences.</p>
  <h1 id="references" class="unnumbered">References</h1>
  <div id="refs" class="references">
  <div id="ref-bioconda">
  <p>&#8220;Bioconda.&#8221; n.d. <a href="https://bioconda.github.io/" class="uri">https://bioconda.github.io/</a>.</p>
  </div>
  <div id="ref-DiTommaso2017">
  <p>Di Tommaso, Paolo, Maria Chatzou, Evan W Floden, Pablo Prieto Barja, Emilio Palumbo, and Cedric Notredame. 2017. &#8220;Nextflow enables reproducible computational workflows.&#8221; <em>Nature Biotechnology</em> 35 (4): 316&#8211;19. doi:<a href="https://doi.org/10.1038/nbt.3820">10.1038/nbt.3820</a>.</p>
  </div>
  <div id="ref-Koster2012a">
  <p>Koster, J., and S. Rahmann. 2012. &#8220;Snakemake&#8211;a scalable bioinformatics workflow engine.&#8221; <em>Bioinformatics</em> 28 (19): 2520&#8211;2. doi:<a href="https://doi.org/10.1093/bioinformatics/bts480">10.1093/bioinformatics/bts480</a>.</p>
  </div>
  <div id="ref-pip">
  <p>&#8220;Pip.&#8221; n.d. <a href="https://pip.pypa.io/en/stable/" class="uri">https://pip.pypa.io/en/stable/</a>.</p>
  </div>
  <div id="ref-Vivian2017">
  <p>Vivian, John, Arjun Arkal Rao, Frank Austin Nothaft, Christopher Ketchum, Joel Armstrong, Adam Novak, Jacob Pfeil, et al. 2017. &#8220;Toil enables reproducible, open source, big biomedical data analyses.&#8221; <em>Nature Biotechnology</em> 35 (4): 314&#8211;16. doi:<a href="https://doi.org/10.1038/nbt.3772">10.1038/nbt.3772</a>.</p>
  </div>
  </div>
  </div>
</div>
EOF
