body = <<-EOF
<meta name="citation_title" content="A framework for generating interactive reports for cancer genome analysis">
<meta name="citation_author" content="Okada, Ai"><meta name="citation_author" content="Chiba, Kenichi"><meta name="citation_author" content="Tanaka, Hiroko"><meta name="citation_author" content="Miyano, Satoru"><meta name="citation_author" content="Shiraishi, Yuichi">
<meta name="citation_publication_date" content="2017//">
<meta name="citation_journal_title" content="The Journal of Open Source Software">
<meta name="citation_pdf_url" content="http://www.theoj.org/joss-papers/joss.00457/10.21105.joss.00457.pdf">
<meta name="citation_doi" content="10.21105/joss.00457">
<meta name="citation_issn" content="2475-9066">
<div class="accepted-paper">
  <h1>A framework for generating interactive reports for cancer genome analysis</h1>
  <div class="columns links">
    <div class="column four-fifths" style="padding-bottom: 10px;">
      <strong>Authors</strong>
      <ul class="author-list">
            <li><a href="http://orcid.org/0000-0002-9782-5900" target="_blank">Ai Okada</a></li>
            <li><a href="http://orcid.org/" target="_blank">Kenichi Chiba</a></li>
            <li><a href="http://orcid.org/" target="_blank">Hiroko Tanaka</a></li>
            <li><a href="http://orcid.org/" target="_blank">Satoru Miyano</a></li>
            <li><a href="http://orcid.org/0000-0001-6144-5845" target="_blank">Yuichi Shiraishi</a></li>
            </ul>
    </div>
    <div class="one-third column">
      <span class="repo">Repository:<br /><a href="https://github.com/Genomon-Project/paplot/">Repository link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Paper:<br /><a href="http://www.theoj.org/joss-papers/joss.00457/10.21105.joss.00457.pdf">PDF link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Review:<br /><a href="https://github.com/openjournals/joss-reviews/issues/457">View review issue &raquo;</a></span>
    </div>

    <div class="one-third column" style="padding-top: 20px;">
      <span class="repo">DOI:<br /><a href="https://doi.org/10.21105/joss.00457">https://doi.org/10.21105/joss.00457</a></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Status badge:<br /><img src="http://joss.theoj.org/papers/10.21105/joss.00457/status.svg"></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">
        Submitted: 13 November 2017 <br />
        Published: 06 December 2017
      </span>

    </div>
    <div class="two-thirds column" style="padding-top: 20px;">
      <span class="paper">Citation:<br />
      <small>Okada et al., (2017). A framework for generating interactive reports for cancer genome analysis. <em>Journal of Open Source Software</em>, 2(20), 457, doi:10.21105/joss.00457</small>
    </div>
  </div>
  <div class="paper-body">
  <h1 id="summary">Summary</h1>
  <p>We introduce paplot, the software for generating dynamic reports that are frequently necessary in the post analyticalphases of cancer genome studies. The &quot;interactive&quot; nature of the paplot-generated reportsenables users to extract much richer information than that obtained from static graphs via most conventional visualization tools.</p>
  <p>The worldwide rapid accumulation of massive cancer genome sequencing data presents an unprecedented opportunity for obtain-ing enormous biologically and clinically important information. Several analytical tools have been developed for identifying somat-ic mutations <span class="citation">(Koboldt et al. 2012; Shiraishi et al. 2013)</span>, struc-tural variations <span class="citation">(Rausch et al. 2012)</span>, and prominent mutation signatures <span class="citation">(Shiraishi et al. 2015)</span> and so on. However, the subse-quent visualization and interpretation often demand us additional laborious efforts such as formatting and organizing the result ta-bles via additional scripting, as well as graphic manipulation via drawing tools, occasionally with multiple trial-and-error parameter adjustments. Additionally, as many existing tools only provide static figures <span class="citation">(Skidmore et al. 2016; Yin, Cook, and Lawrence 2012)</span>, it is neces-sary to again repeatedly execute drawing tools with different set-tings even for generating slightly different visualizations (e.g., sorting and filtering by specific criteria). Furthermore, even when some interesting characteristics are observed, we often need to bother to refer back to the raw analytical result tables for identify-ing the samples and the exact values of interest, as these may not be shown in the static figures. Here, we present <a href="https://github.com/Genomon-Project/paplot">paplot</a>, a suite of programs to generate various reports for commonly needed tasks in cancer genome analysis. paplot accepts text-based input data and configuration files and generates HTML files with dynamic, interactive data visualizations powered by <a href="https://d3js.org">D3.js</a>. By modifying settings in the configuration files, users can freely change various visual aspects such as colors of objects and contents of pop-up windows appearing after DOM events such as mouse-click and mouse-over.</p>
  <p>Interactive graphics contain richer information than static ones, thus facilitating effective and intuitive interpretation of analytical results. Owing to the advancement in information technologies and development of application frameworks for easy generation of interactive graphics, using interactive graphics for academic re-search and auto-generated reports for clinical sequencing will be-come more common. We believe that paplot will be useful for academic research and clinical practice.</p>
  <hr />
  <div class="figure">
  <img src="graphs.png" alt="Example of interactive reports created by paplot. (A) Quality Control Report. (B) Chromosomal Aberration Report. (C) Mutation Matrix Report. (D) Mutational Signature Report." />
  <p class="caption">Example of interactive reports created by paplot. (A) Quality Control Report. (B) Chromosomal Aberration Report. (C) Mutation Matrix Report. (D) Mutational Signature Report.</p>
  </div>
  <h2 id="quality-control-report">Quality Control Report</h2>
  <p>Confirming the quality of sequence data (e.g., coverage, insert size and mapping ratio) is a fundamental step in sequence analysis. In many cases, the quality control check is performed by visual in-spection of the barplot to check whether there are some samples with strangely heights. However, as adding the X-axis labels for all the samples is not always possible especially when the sample size in the cohort is large, we have to spend excessive time to retrieve the raw table data to identify the samples with the strange features. By using the Quality Control Report generated by paplot (Figure 1A), one can immediately identify such samples via pop-up con-tents.</p>
  <h2 id="chromosomal-aberration-report">Chromosomal Aberration Report</h2>
  <p>Chromosomal aberrations play important roles in cancer develop-ment <span class="citation">(Mitelman, Johansson, and Mertens 2007)</span>. After characterizing chromosomal aberrations (genomic structural variations from whole genome sequencing data or fusion transcripts from transcriptome sequenc-ing data), it is common to generate a visual overview of chromo-somal aberrations, typically using the circos plot <span class="citation">(Krzywinski et al. 2009)</span>, to examine for certain specific phenomena such as chromothripsis <span class="citation">(P. J. Stephens et al. 2011)</span>. When the users open the Chromosomal Aberrations Report (Figure 1B), they will first find thumbnails for chromosomal aber-rations across all the samples, thus providing a cohort-level land-scape. On clicking one of the thumbnails, the bigger picture of chromosomal aberrations for the sample of interest appears, and detailed information of each variant (e.g., exact coordinates of breakpoints and affected genes) pops up upon mouse-over. Fur-thermore, the distribution for the cumulative count of breakpoints of chromosomal aberrations partitioned by a certain chromosomal bin size (default size is 50 Kbp) is automatically generated, facili-tating identification of common cancer-driving variants represented by prominent peaks.</p>
  <h2 id="mutation-matrix-report">Mutation Matrix Report</h2>
  <p>Obtaining a comprehensive vision of somatic mutations across cancer genomes in a cohort of interest to examine frequencies of affected genes, relative ratio of variant types (e.g., missense, non-sense), and co-occurrence and exclusivity between two genes, is a principal goal in cancer genome analysis. paplot produces the Mu-tation Matrix Report (Figure 1C), a dynamic report illustrating the mutation status, which can fulfill the above purpose. Users can enjoy sorting by several criteria such as sample-wise and gene-wise frequencies. In addition, the user can generate a &#8220;waterfall&#8221; plot <span class="citation">(Skidmore et al. 2016)</span>, where samples are hierarchically divided per the mutation status of genes in descending order of mutation frequencies. Furthermore, clinical information can be included in the plot footer, which can work as additional sort keys.</p>
  <h2 id="mutational-signature-report">Mutational Signature Report</h2>
  <p>In the past few years, massive genome cancer sequencing analysis revealed many characteristic patterns of somatic mutations. The profile of such mutation signature is typically represented by a bar plot with 96 possible types (6 types of substitution &#215; 4 types of 5' base &#215; 4 types of 3' base) <span class="citation">(Alexandrov et al. 2013)</span>. We have recently proposed another way of visualizing mutation signatures to enable the inclusion of more contextual factors <span class="citation">(Shiraishi et al. 2015)</span>. paplot can generate the Mutational Signature Report (Figure 1D) in both format types. Furthermore, since the barplot of mem-bership parameters (estimated fractions of operative mutation signatures for each sample) is also generated, the user can easily spot the sample with prominent mutation signatures via pop-up windows.</p>
  <h1 id="funding">Funding</h1>
  <p>This work was supported by a Grant-in-Aid from the Japan Agency for Medical Research and Development [Advanced Genome Research and Bioinformatics Study to Facilitate Medical Innovation (17km0405207h0002)] and a Grant-in-Aid for Scientific Research (KAKENHI 15H05912 and 15K00398).</p>
  <h1 id="references" class="unnumbered">References</h1>
  <div id="refs" class="references">
  <div id="ref-alexandrov2013deciphering">
  <p>Alexandrov, Ludmil B, Serena Nik-Zainal, David C Wedge, Peter J Campbell, and Michael R Stratton. 2013. &#8220;Deciphering Signatures of Mutational Processes Operative in Human Cancer.&#8221; <em>Cell Reports</em> 3 (1). Elsevier: 246&#8211;59. doi:<a href="https://doi.org/10.1016/j.celrep.2012.12.008">10.1016/j.celrep.2012.12.008</a>.</p>
  </div>
  <div id="ref-koboldt2012varscan">
  <p>Koboldt, Daniel C, Qunyuan Zhang, David E Larson, Dong Shen, Michael D McLellan, Ling Lin, Christopher A Miller, Elaine R Mardis, Li Ding, and Richard K Wilson. 2012. &#8220;VarScan 2: Somatic Mutation and Copy Number Alteration Discovery in Cancer by Exome Sequencing.&#8221; <em>Genome Research</em> 22 (3). Cold Spring Harbor Lab: 568&#8211;76. doi:<a href="https://doi.org/10.1101/gr.129684.111">10.1101/gr.129684.111</a>.</p>
  </div>
  <div id="ref-krzywinski2009circos">
  <p>Krzywinski, Martin, Jacqueline Schein, Inanc Birol, Joseph Connors, Randy Gascoyne, Doug Horsman, Steven J Jones, and Marco A Marra. 2009. &#8220;Circos: An Information Aesthetic for Comparative Genomics.&#8221; <em>Genome Research</em> 19 (9). Cold Spring Harbor Lab: 1639&#8211;45. doi:<a href="https://doi.org/10.1101/gr.092759.109">10.1101/gr.092759.109</a>.</p>
  </div>
  <div id="ref-mitelman2007impact">
  <p>Mitelman, Felix, Bertil Johansson, and Fredrik Mertens. 2007. &#8220;The Impact of Translocations and Gene Fusions on Cancer Causation.&#8221; <em>Nature Reviews Cancer</em> 7 (4). Nature Publishing Group: 233&#8211;45. doi:<a href="https://doi.org/10.1038/nrc2091">10.1038/nrc2091</a>.</p>
  </div>
  <div id="ref-rausch2012delly">
  <p>Rausch, Tobias, Thomas Zichner, Andreas Schlattl, Adrian M St&#252;tz, Vladimir Benes, and Jan O Korbel. 2012. &#8220;DELLY: Structural Variant Discovery by Integrated Paired-End and Split-Read Analysis.&#8221; <em>Bioinformatics</em> 28 (18). Oxford University Press: i333&#8211;i339. doi:<a href="https://doi.org/10.1093/bioinformatics/bts378">10.1093/bioinformatics/bts378</a>.</p>
  </div>
  <div id="ref-shiraishi2013empirical">
  <p>Shiraishi, Yuichi, Yusuke Sato, Kenichi Chiba, Yusuke Okuno, Yasunobu Nagata, Kenichi Yoshida, Norio Shiba, et al. 2013. &#8220;An Empirical Bayesian Framework for Somatic Mutation Detection from Cancer Genome Sequencing Data.&#8221; <em>Nucleic Acids Research</em> 41 (7). Oxford University Press: e89&#8211;e89. doi:<a href="https://doi.org/10.1093/nar/gkt126">10.1093/nar/gkt126</a>.</p>
  </div>
  <div id="ref-shiraishi2015simple">
  <p>Shiraishi, Yuichi, Georg Tremmel, Satoru Miyano, and Matthew Stephens. 2015. &#8220;A Simple Model-Based Approach to Inferring and Visualizing Cancer Mutation Signatures.&#8221; <em>PLoS Genetics</em> 11 (12). Public Library of Science: e1005657. doi:<a href="https://doi.org/10.1371/journal.pgen.1005657">10.1371/journal.pgen.1005657</a>.</p>
  </div>
  <div id="ref-skidmore2016genvisr">
  <p>Skidmore, Zachary L, Alex H Wagner, Robert Lesurf, Katie M Campbell, Jason Kunisaki, Obi L Griffith, and Malachi Griffith. 2016. &#8220;GenVisR: Genomic Visualizations in R.&#8221; <em>Bioinformatics</em> 32 (19). Oxford University Press: 3012&#8211;4. doi:<a href="https://doi.org/10.1093/bioinformatics/btw325">10.1093/bioinformatics/btw325</a>.</p>
  </div>
  <div id="ref-stephens2011massive">
  <p>Stephens, Philip J, Chris D Greenman, Beiyuan Fu, Fengtang Yang, Graham R Bignell, Laura J Mudie, Erin D Pleasance, et al. 2011. &#8220;Massive Genomic Rearrangement Acquired in a Single Catastrophic Event During Cancer Development.&#8221; <em>Cell</em> 144 (1). Elsevier: 27&#8211;40. doi:<a href="https://doi.org/10.1016/j.cell.2010.11.055">10.1016/j.cell.2010.11.055</a>.</p>
  </div>
  <div id="ref-yin2012ggbio">
  <p>Yin, Tengfei, Dianne Cook, and Michael Lawrence. 2012. &#8220;Ggbio: An R Package for Extending the Grammar of Graphics for Genomic Data.&#8221; <em>Genome Biology</em> 13 (8). BioMed Central: R77. doi:<a href="https://doi.org/10.1186/gb-2012-13-8-r77">10.1186/gb-2012-13-8-r77</a>.</p>
  </div>
  </div>
  </div>
</div>
EOF
