body = <<-EOF
<meta name="citation_title" content="malan: <strong>MA</strong>le <strong>L</strong>ineage <strong>AN</strong>alysis">
<meta name="citation_author" content="Meyer Andersen, Mikkel">
<meta name="citation_publication_date" content="2018//">
<meta name="citation_journal_title" content="The Journal of Open Source Software">
<meta name="citation_pdf_url" content="http://www.theoj.org/joss-papers/joss.00684/10.21105.joss.00684.pdf">
<meta name="citation_doi" content="10.21105/joss.00684">
<meta name="citation_issn" content="2475-9066">
<div class="accepted-paper">
  <h1>malan: <strong>MA</strong>le <strong>L</strong>ineage <strong>AN</strong>alysis</h1>
  <div class="columns links">
    <div class="column four-fifths" style="padding-bottom: 10px;">
      <strong>Authors</strong>
      <ul class="author-list">
            <li><a href="http://orcid.org/0000-0002-0234-0266" target="_blank">Mikkel Meyer Andersen</a></li>
            </ul>
    </div>
    <div class="one-third column">
      <span class="repo">Repository:<br /><a href="https://github.com/mikldk/malan">Repository link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Paper:<br /><a href="http://www.theoj.org/joss-papers/joss.00684/10.21105.joss.00684.pdf">PDF link &raquo;</a></span>
    </div>
    <div class="one-third column">
      <span class="paper">Review:<br /><a href="https://github.com/openjournals/joss-reviews/issues/684">View review issue &raquo;</a></span>
    </div>

    <div class="one-third column" style="padding-top: 20px;">
      <span class="repo">DOI:<br /><a href="https://doi.org/10.21105/joss.00684">https://doi.org/10.21105/joss.00684</a></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">Status badge:<br /><img src="http://joss.theoj.org/papers/10.21105/joss.00684/status.svg"></span>
    </div>
    <div class="one-third column" style="padding-top: 20px;">
      <span class="paper">
        Submitted: 16 April 2018 <br />
        Published: 07 May 2018
      </span>

    </div>
    <div class="two-thirds column" style="padding-top: 20px;">
      <span class="paper">Citation:<br />
      <small>Andersen, (2018). malan: **MA**le **L**ineage **AN**alysis. <em>Journal of Open Source Software</em>, 3(25), 684. https://doi.org/10.21105/joss.00684</small>
    </div>
  </div>
  <div class="paper-body">
  <h1 id="summary">Summary</h1>
  <p>Forensic DNA profiles from the Y-chromosome are valuable when there is a mixture of male-source and female-source DNA, and interest centres on the identity of the male source(s) of the DNA (as only males have Y-chromosomes). This happens for example when a male has been accused of assaulting a female.</p>
  <p>The problem of evaluating evidential weight is even more challenging for Y-profiles than for autosomal profiles that are based on the non-sex chromosomes. At the core of the weight of evidence for autosomal short tandem repeat (STR) profiles used in forensic genetics is the <em>match probability</em>, which is the conditional probability that a particular individual <span class="math inline"><em>X</em></span> has a matching profile, given that the queried contributor, <span class="math inline"><em>Q</em></span>, has it <span class="citation" data-cites="DJBwoe2">(Steele and Balding 2015)</span>. Matching at a single, autosomal STR allele is relatively common: typically a few percent of individuals from the same population share a given allele. The probability of matching is increased when <span class="math inline"><em>X</em></span> is a relative of <span class="math inline"><em>Q</em></span>, but for typical population sizes most of the individuals sharing a given allele are not closely related to <span class="math inline"><em>Q</em></span>.</p>
  <p>However, unlike for autosomal profiles, Y-profile matches are due to patrilineal relatedness that is typically too remote to be recognized, but close compared with the relatedness of random pairs from the population <span class="citation" data-cites="AndersenPLOSGEN2017">(Andersen and Balding 2017)</span>. This was described by <span class="citation" data-cites="AndersenPLOSGEN2017">(Andersen and Balding 2017)</span> that also propose a way to interpret a matching Y-chromosomal profile given these properties. The proposed interpretation was based on the distribution of the number of males with Y-profile matching that of the queried contributor <span class="math inline"><em>Q</em></span>. Analyses in <span class="citation" data-cites="AndersenPLOSGEN2017">(Andersen and Balding 2017)</span> were performed by a simulation model to approximate the distribution of the number of men in a population with matching Y-profiles. Key parameters of the model include the STR locus mutation rates, the variance in reproductive success <span class="citation" data-cites="AndersenPLOSGEN2017">(Andersen and Balding 2017)</span> (VRS), and the population growth rate. The simulation model was implemented and made available in the easy-to-use open-source software presented in this paper: <code>malan</code> (<strong>MA</strong>le <strong>L</strong>ineage <strong>AN</strong>alysis).</p>
  <p>The <code>malan</code> software is made available as an R <span class="citation" data-cites="R">(R Core Team 2018)</span> package with extensive use of C++ for efficient computations via Rcpp <span class="citation" data-cites="Rcpp">(Eddelbuettel and Balamuta 2017)</span>. This software was used for the analyses performed in <span class="citation" data-cites="AndersenPLOSGEN2017">(Andersen and Balding 2017)</span>. The simulation model allows for flexible simulations by first simulating a genealogy (with population sizes at each generation specified by a vector) with different parameters as described by <span class="citation" data-cites="AndersenPLOSGEN2017">(Andersen and Balding 2017)</span>. A forensic Y-chromosome profile typically consists of the allele at between 15 and 30 STR loci <span class="citation" data-cites="Butler05">(Butler et al. 2005)</span> and is often referred to as a Y-STR haplotype. In the simulated genealogy, the <code>malan</code> software makes it possible to impose Y-STR haplotypes in different ways.</p>
  <p>The <code>malan</code> software makes it possible to query the population in multiple ways. For example to count the number of males in the population with a certain Y-STR haplotype. Or obtain the distribution of number of meioses between a queried contributor and the individuals in the population with a matching Y-STR haplotype.</p>
  <p>The documentation of <code>malan</code> consists of manual pages for the various available functions, articles describing how to perform contiguous analyses (<em>vignettes</em>), and unit tests.</p>
  <p>I would like to thank David J Balding for helpful discussions.</p>
  <p>-<img src="paper-fig-simulation.png" alt="Simulation illustration and example of analysis of matching individuals." /></p>
  <h1 id="references" class="unnumbered">References</h1>
  <div id="refs" class="references">
  <div id="ref-AndersenPLOSGEN2017">
  <p>Andersen, MM, and DJ Balding. 2017. &#8220;How convincing is a matching Y-chromosome profile?&#8221; <em>PLOS Genetics</em> 13 (11):e1007028. <a href="https://doi.org/10.1371/journal.pgen.1007028" class="uri">https://doi.org/10.1371/journal.pgen.1007028</a>.</p>
  </div>
  <div id="ref-Butler05">
  <p>Butler, JM, AE Decker, MC Kline, and PM Vallone. 2005. &#8220;Chromosomal Duplications Along the Y Chromosome and Their Potential Impact on Y-STR Interpretation.&#8221; <em>Journal of Forensic Sciences</em> 50 (4):853&#8211;9. <a href="https://doi.org/10.1520/JFS2004481" class="uri">https://doi.org/10.1520/JFS2004481</a>.</p>
  </div>
  <div id="ref-Rcpp">
  <p>Eddelbuettel, D, and JJ Balamuta. 2017. &#8220;Extending <em>R</em> with <em>C++</em>: A Brief Introduction to <em>Rcpp</em>.&#8221; <em>PeerJ Preprints</em> 5 (August):e3188v1. <a href="https://doi.org/10.7287/peerj.preprints.3188v1" class="uri">https://doi.org/10.7287/peerj.preprints.3188v1</a>.</p>
  </div>
  <div id="ref-R">
  <p>R Core Team. 2018. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
  </div>
  <div id="ref-DJBwoe2">
  <p>Steele, CD, and DJ Balding. 2015. <em>Weight of Evidence for Forensic DNA Profiles</em>. 2nd ed. Wiley. <a href="https://doi.org/10.1002/9780470867693" class="uri">https://doi.org/10.1002/9780470867693</a>.</p>
  </div>
  </div>
  </div>
</div>
EOF
